pymongo/errors.py

```diff
@@ -167,6 +167,11 @@ def details(self):
         """
         return self.__details
 
+    def __str__(self):
+        output_str = "%s, full error: %s" % (self._message, self.__details)
+        if isinstance(output_str, unicode):
```

Comment:
Since `unicode` does not exist in Python 3 we'll need to restrict this to Python 2 only (`sys.version_info[0] == 2`).